pkg/cmd/cli/secrets/sshauth.go

```diff
@@ -0,0 +1,194 @@
+package secrets
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	// CreateSSHAuthSecretRecommendedCommandName represents name of subcommand for `oc secrets` command
+	CreateSSHAuthSecretRecommendedCommandName = "new-sshauth"
+
+	createSSHAuthSecretLong = `
+Create a new SSH authentication secret
+
+SSH authentication secrets are used to authenticate against SCM servers.
+
+When creating applications, you may have a SCM server that requires SSH authentication - private SSH key.
+In order for the nodes to clone source code on your behalf, they have to have the credentials. You can 
+provide this information by creating a 'sshauth' secret and attaching it to your service account.`
+
+	createSSHAuthSecretExample = `  // If your SSH authentication method requires only private SSh key, add it by using:
+  $ %[1]s SECRET_NAME --ssh-privatekey=FILENAME
+
+  // If your SSH authentication method requires also CA certificate, add it by using:
+  $ %[1]s SECRET_NAME --ssh-privatekey=FILENAME --ca-cert=FILENAME
+
+  // If you do already have a .gitconfig file needed for authentication, you can create a gitconfig secret by using:
+  $ %[2]s SECRET_NAME path/to/.gitconfig`
+)
+
+// CreateSSHAuthSecretOptions holds the credential needed to authenticate against SCM servers.
+type CreateSSHAuthSecretOptions struct {
+	SecretName      string
+	PrivateKeyPath  string
+	CertificatePath string
+	GitConfigPath   string
+
+	PromptForPassword bool
+
+	Out io.Writer
+
+	SecretsInterface client.SecretsInterface
+}
+
+// NewCmdCreateSSHAuthSecret implements the OpenShift cli secrets new-sshauth subcommand
+func NewCmdCreateSSHAuthSecret(name, fullName string, f *kcmdutil.Factory, out io.Writer, newSecretFullName, ocEditFullName string) *cobra.Command {
+	o := &CreateSSHAuthSecretOptions{
+		Out: out,
+	}
+
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s SECRET_NAME --ssh-privatekey=FILENAME [--ca-cert=FILENAME --gitconfig=FILENAME]", name),
```

Comment:
`[--ca-cert=FILENAME --gitconfig=FILENAME]` should be `[--ca-cert=FILENAME] [--gitconfig=FILENAME]